plugins/postcss-cascade-layers/src/desugar-nested-layers.ts

```diff
@@ -0,0 +1,61 @@
+import type { Container, AtRule, ChildNode } from 'postcss';
```

Comment:
```css
@layer A {
  @layer B {
    .foo { color: green; }
  }
}
```

Is equivalent to :

```css
@layer A.B {
  .foo { color: green; }
}
```

Desugaring to `@layer A.B` simplifies further processing.